homeassistant/components/history_stats/sensor.py

```diff
@@ -271,7 +271,8 @@ def update_period(self):
             except (TemplateError, TypeError) as ex:
                 HistoryStatsHelper.handle_template_exception(ex, "start")
                 return
-            start = dt_util.parse_datetime(str(start_rendered))
+            if isinstance(start_rendered, str):
```

Comment:
If it's not a string, we should throw an error and return. Or else it will fail on line 309. Same for end.

Reply:
If it is not a string, it will try the approach directly below this one (which is timestamp based), which throws the error in that case.

In case nothing works, it will never reach line 309.